query-connector/src/app/ui/components/withAuth/WithAuth.tsx

```diff
@@ -29,7 +29,12 @@ const WithAuth: React.FC<React.PropsWithChildren> = ({ children }) => {
     !access.includes(session?.user?.role as RoleTypeValues)
   ) {
     redirect("/unauthorized");
+  } else if (session === null) {
```

Comment:
think we might be able to use [the `status` variable](https://next-auth.js.org/getting-started/client#example) (which correspond to the different values of session here) that next-auth gives us in order to make these checks more readable

Reply:
Thanks! I will work on this update